query/src/provider/overlap.rs

```diff
@@ -968,4 +1012,17 @@ mod test {
         }
         s
     }
+
+    // convert from Vec<Vec<Arc<TestChunk>>> to Vec<Vec<Arc<dyn QueryChunk>>>
+    fn to_group_query_chunks(groups: Vec<Vec<Arc<TestChunk>>>) -> Vec<Vec<Arc<dyn QueryChunk>>> {
+        groups
+            .iter()
+            .map(|chunks| {
+                chunks
+                    .iter()
+                    .map(|c| Arc::clone(c) as Arc<dyn QueryChunk>)
+                    .collect::<Vec<_>>()
+            })
+            .collect::<Vec<_>>()
```

Comment:
I wonder if you could do something like this (untested):

```suggestion
        groups
            .into_iter()
            .map(|chunks| {
                chunks
                    .into_iter()
                    .map(|c| as _)
                    .collect::<Vec<_>>()
            })
            .collect::<Vec<_>>()
```

Reply:
It works. Thanks @alamb